app/viewModels/P5/departure/IntentionToControlP5ViewModel.scala

```diff
@@ -69,24 +72,23 @@ case class IntentionToControlP5ViewModel(sections: Seq[Section], requestedDocume
 
 object IntentionToControlP5ViewModel {
 
-  class IntentionToControlP5ViewModelProvider @Inject() () {
+  class IntentionToControlP5ViewModelProvider @Inject() (referenceDataService: ReferenceDataService) {
 
     def apply(
       ie060: CC060CType,
       customsOffice: CustomsOffice
-    )(implicit messages: Messages): IntentionToControlP5ViewModel = {
-      val helper = new IntentionToControlP5MessageHelper(ie060)
+    )(implicit messages: Messages, ec: ExecutionContext, hc: HeaderCarrier): Future[IntentionToControlP5ViewModel] = {
+      val helper = new IntentionToControlP5MessageHelper(ie060, referenceDataService)
 
       val requestedDocuments: Boolean = ie060.informationRequested
       val lrn                         = ie060.TransitOperation.LRN
 
-      val intentionToControlSection = helper.buildIntentionToControlSection()
-
-      val sections = Seq(intentionToControlSection) ++ helper.documentSection()
-
-      new IntentionToControlP5ViewModel(sections, requestedDocuments, lrn, customsOffice: CustomsOffice)
+      for {
+        intentionToControlSection <- helper.buildIntentionToControlSection()
```

Comment:
Looks like we have access to the customsOffice here but then we're fetching it again in the helper? Could we just reuse this one?